src/NuGet.Core/NuGet.Versioning/VersionRangeFactory.cs

```diff
@@ -233,6 +236,19 @@ public static bool TryParse(string value, bool allowFloating, out VersionRange v
                 }
             }
 
+            if (minVersion != null && maxVersion != null
+                && minVersion >= maxVersion
+                && !(minVersion == maxVersion && isMinInclusive && isMaxInclusive))
+            {
+                if ((partsLength == 1 && minVersion == maxVersion
+                            && (isMinInclusive ^ isMaxInclusive || !isMinInclusive && !isMaxInclusive))
```

Comment:
I'm having a hard time reading this statement. 

Can you try doing another pass on simplifying this logic please? 

For example: `isMinInclusive ^ isMaxInclusive || !isMinInclusive && !isMaxInclusive` can be rewritten as `!(isMinInclusive && isMaxInclusive)` 

Given the complexity of the checks, I think some comments would be helpful.

Reply:
Ok. I simplified the logic and made it easy to reason with small parts. Please check.